### PR TITLE
Use AndroidX in new projects by default

### PR DESCRIPTION
**Pull request by blasten**
## Description

Create new AndroidX projects by default.
This can be merged after https://github.com/flutter/flutter/pull/40810

## Related Issues

https://flutter.dev/go/androidx-transition

## Tests

I added the following tests: ensure that Flutter creates an AndroidX project by default.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by xster** [comment]
LGTM. You wanna manually move all the projects in the Flutter repo (like the tests and examples and stuff) as well?

**Comment by blasten** [comment]
@xster yes. I have a separate PR that will change that for existing examples and tests.

File: dev/integration_tests/android_host_app/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,7 +4,8 @@
     package="io.flutter.add2app">
 
     <application android:allowBackup="false"
-        tools:ignore="GoogleAppIndexingWarning,MissingApplicationIcon">
+        tools:ignore="GoogleAppIndexingWarning,MissingApplicationIcon"
+        tools:replace="android:appComponentFactory">
```
Comment (xster) [inline dev/integration_tests/android_host_app/app/src/main/AndroidManifest.xml]: What does this do?

**Review by xster — COMMENTED** [review]
Still LGTM

File: dev/integration_tests/android_host_app/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (blasten) [reply dev/integration_tests/android_host_app/app/src/main/AndroidManifest.xml]: committed this by mistake. Thanks for catching it up.

**Pull request merged**